src/musicbrainz/tagfetcher.cpp

```diff
@@ -343,25 +349,37 @@ void TagFetcher::slotCoverArtArchiveLinksTaskAborted() {
         // stray call from an already aborted try
         return;
     }
-
     terminate();
 }
 
 void TagFetcher::slotCoverArtArchiveLinksTaskNetworkError(
         QNetworkReply::NetworkError errorCode,
         const QString& errorString,
+
         const mixxx::network::WebResponseWithContent& responseWithContent) {
-    // TODO(XXX) Handle the error better for Cover Art Archive Links.
-    Q_UNUSED(errorCode);
-    Q_UNUSED(errorString);
     Q_UNUSED(responseWithContent);
     DEBUG_ASSERT_QOBJECT_THREAD_AFFINITY(this);
     if (m_pCoverArtArchiveLinksTask.get() != sender()) {
-        // stray call from an already aborted try
-        return;
+        return; // Stray call from an already aborted try
+    }
+    m_pCoverArtArchiveLinksTask = make_parented<mixxx::CoverArtArchiveLinksTask>(
+            &m_network,
+            std::move(errorString),
+            this);
```

Comment:
I don't understand the purpose of this.

In this function we should actually only need to interpret the errorCode as you do below, then pass the user string to DlgTagfetcher.